### PR TITLE
In physical addressing, functions can return pointers

### PR DESCRIPTION
**Pull request by dneto0**
Fixes https://github.com/KhronosGroup/SPIRV-Tools/issues/229

### CODE REVIEW

File: source/validate_id.cpp [hunk 1]
```diff
@@ -1710,10 +1710,12 @@ bool idUsage::isValid<SpvOpReturnValue>(const spv_instruction_t* inst,
...
-  if (SpvOpTypePointer == valueType.second.opcode) {
+  if (addressingModel == SpvAddressingModelLogical &&
+      SpvOpTypePointer == valueType.second.opcode) {
     DIAG(valueIndex) << "OpReturnValue value's type <id> '"
                      << value.second.type_id
-                     << "' is a pointer, but a pointer can only be an operand "
+                     << "' is a pointer, but in the Logical addressing model "
+                        "a pointer can only be an operand "
                         "to OpLoad, OpStore, OpAccessChain, or "
                         "OpInBoundsAccessChain.";
```
Comment (dekimir) [inline source/validate_id.cpp]: This is no longer accurate: https://www.khronos.org/registry/spir-v/specs/1.1/SPIRV.html#_universal_validation_rules

**Comment by dekimir** [comment]
-1 for a minor error-message inaccuracy, otherwise looks good.

**Comment by dneto0** [comment]
Good catch on the message.
All feedback addressed.

**Comment by dekimir** [comment]
+2

**Pull request merged**